cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -82,7 +85,7 @@ public BasicMapReduceContext(Program program,
                                MetricsCollectionService metricsCollectionService,
                                DatasetFramework dsFramework,
                                CConfiguration conf) {
-    super(program, runId, runtimeArguments, datasets,
+    super(program, runId, extractScope(program.getName(), runtimeArguments), datasets,
```

Comment:
Do you know whether the runtimeArguments are immutable? If so, then extractScope should probably return an ImmutableMap, too. But I am not sure. 

Rationale: If it is mutable, then whoever receives this may be under the illusion that he can make additions and they get passed on. But that is not that case.